### ISSUE TITLE
ModuleNotFoundError: No module named "taming"

### ISSUE THREAD

**Issue by Nughu** [CompVis/stable-diffusion#162]
When running txt2img.py on Rocm 5.1.1 inside the ldm conda environment, I am running into
ModuleNotFoundError: No module named "taming".

![IMG_20220902_010908](https://user-images.githubusercontent.com/112648460/188027450-5e6a54fb-4229-4e6f-bc9c-41210b07d04c.jpg)

**Comment by hx3333** [comment]
I got the same issue, have you found any solutions?@Nughu

**Comment by davidusb-geek** [comment]
The packaging of this code is just a mess. A good requirements.txt is missing.
To solve this issue you just need to clone this repo: https://github.com/CompVis/taming-transformers.git
And the add the path of the cloned repo to your PYTHONPATH.
In Linux this will be: export PYTHONPATH="${PYTHONPATH}:/home/user/path/to/cloned/repo"
And that's it, your system should be able to recognize "taming" as a proper Python module

**Comment by Nughu** [comment]
> I got the same issue, have you found any solutions?@Nughu 

I did, by downloading taming-transformers, moving the "taming" folder to the stable diffusion directory, and running pip install -e . Inside the stable diffusion directory with conda environment activated.

**Comment by davidusb-geek** [comment]
You should not move the taming folder and there is no need to install anything. Just add the taming module to the PYTHONPATH

**Comment by hniksoleimani** [comment]
> You should not move the taming folder and there is no need to install anything. Just add the taming module to the PYTHONPATH

it would have been immensely helpful if the instruction has been given.

**Comment by Tempaccnt** [comment]
Found a solution for windows users, download the taming-transforms files as a zip file. you can find it here https://github.com/CompVis/taming-transformers. or look through the GitHub profile of CompVis if you don't like opening links from strangers.

then extract the files inside the following path: ..../stable-diffusion-webui/repositories\taming-transformers

ps: make sure that you extract the files directly to that path and not extracting them into a folder.

**Comment by FreeBlues** [comment]
@davidusb-geek Thanks! I solved with your solution on my macOS(Intel cpu)!

**Comment by Los-3r** [comment]
> Found a solution for windows users, download the taming-transforms files as a zip file. you can find it here https://github.com/CompVis/taming-transformers. or look through the GitHub profile of CompVis if you don't like opening links from strangers.
> 
> then extract the files inside the following path: ..../stable-diffusion-webui/repositories\taming-transformers
> 
> ps: make sure that you extract the files directly to that path and not extract them into a folder. also, technically this method should work for macOS and Linux too but I haven't tested it on them because I only have a windows machine

Would this work  in the [optimized fork?](https://github.com/basujindal/stable-diffusion)

**Comment by Tempaccnt** [comment]
> > Found a solution for windows users, download the taming-transforms files as a zip file. you can find it here https://github.com/CompVis/taming-transformers. or look through the GitHub profile of CompVis if you don't like opening links from strangers.
> > then extract the files inside the following path: ..../stable-diffusion-webui/repositories\taming-transformers
> > ps: make sure that you extract the files directly to that path and not extract them into a folder. also, technically this method should work for macOS and Linux too but I haven't tested it on them because I only have a windows machine
> 
> Would this work in the [optimized fork?](https://github.com/basujindal/stable-diffusion)

I haven't tested that. but if you get the exact same error, this should work. because the error is basically stating that you are missing the taming module, so it shouldn't matter what system you're on or what version of stable diffusion you are running. but I could be wrong. if you have the optimized fork and are facing the same error please try it out and let us know for sure.

**Comment by mmilunovic-mdcs** [comment]
> Found a solution for windows users, download the taming-transforms files as a zip file. you can find it here https://github.com/CompVis/taming-transformers. or look through the GitHub profile of CompVis if you don't like opening links from strangers.
> 
> then extract the files inside the following path: ..../stable-diffusion-webui/repositories\taming-transformers
> 
> ps: make sure that you extract the files directly to that path and not extract them into a folder. also, technically this method should work for macOS and Linux too but I haven't tested it on them because I only have a windows machine

What's this path "..../stable-diffusion-webui/repositories\taming-transformers"? Is the stable-diffusion root directory?

**Comment by Tempaccnt** [comment]
> 

yes it's the root directory of the stable diffusion were the webui bat file is stored, the dots at the beginning is because it might differ depending on where you choose to install it.

**Comment by wenhaochai** [comment]
> The packaging of this code is just a mess. A good requirements.txt is missing. To solve this issue you just need to clone this repo: https://github.com/CompVis/taming-transformers.git And the add the path of the cloned repo to your PYTHONPATH. In Linux this will be: export PYTHONPATH="${PYTHONPATH}:/home/user/path/to/cloned/repo" And that's it, your system should be able to recognize "taming" as a proper Python module

Thank you so much, that's work for me. BTW, u guys can del the useless .ipynb, .git, and /data files.
